cvat-core/src/server-proxy.ts

```diff
@@ -1074,15 +1018,16 @@ async function restoreTask(storage: Storage, file: File | string) {
                 headers: { 'Upload-Finish': true },
             });
     }
-    return wait();
+    const rqId = response.data.rq_id;
```

Comment:
Here we also do not really need ``if ... else`` anymore

Just: `if (isCloudStorage) { response = ..; return response.data.rq_id }` 

By the way I noticed that requests are not try..catched here.